resources/build/ci/trigger-build-bot.inc.sh

```diff
@@ -66,35 +66,62 @@ function build_bot_check_messages() {
   # Extract the Build-bot commands from commit messages and the PR body
 
 
+  local buildBotShas=($(echo "$prcommits" | "${JQ}" -r '.[].sha'))
   IFS=$'\n'
-  local buildBotCommands=($(echo "$prcommits" | "${JQ}" -r '.[].commit.message' | grep 'Build-bot:' | cut -c 11- -))
   local prCommands=($(echo "$prinfo" | "${JQ}" -r '.body' | tr -d '\r' | grep 'Build-bot:' | cut -c 11- -))
   unset IFS
 
   # The PR body Build-bot comment will be read last, which allows it to override
   # all previous commands
 
-  if [[ ${#prCommands[@]} -gt 0 ]]; then
-    buildBotCommands+=("${prCommands[@]}")
-  fi
+  local buildBotCommand
+  local sha
+  for sha in "${buildBotShas[@]}"; do
 
-  for buildBotCommand in "${buildBotCommands[@]}"; do
-    # Block illegal Build-bot: commands
-    if [[ ! "$buildBotCommand" =~ ^[a-z_,\ :,]+$ ]]; then
-      builder_echo warning "WARNING[Build-bot]: ignoring invalid command [2]: '${buildBotCommand}'"
-      continue
-    fi
+    IFS=$'\n'
+    local buildBotCommands=($(echo "$prcommits" | "${JQ}" -r '.[] | select(.sha | contains("'$sha'")) | .commit.message' | grep 'Build-bot:' | cut -c 11- -))
+    unset IFS
 
-    # debug_echo "buildBotCommand:{$buildBotCommand}"
+    for buildBotCommand in "${buildBotCommands[@]}"; do
+      buildBotCommand=$(builder_trim "${buildBotCommand}")
+      builder_echo heading "Build-bot: Found command in commit ${sha}: '${buildBotCommand}'"
 
-    # We now know that our command has only a-z, comma, colon, and space, so we
-    # can parse without risking escaping our bash jail
+      # Block illegal Build-bot: commands
+      if [[ ! "$buildBotCommand" =~ ^[a-z_,\ :,]+$ ]]; then
+        builder_echo warning "WARNING[Build-bot]: ignoring invalid command: '${buildBotCommand}'"
+        continue
+      fi
 
-    if [[ ! -z "${buildBotCommand// }" ]]; then
-      build_bot_update_commands $buildBotCommand
-    fi
+      # We now know that our command has only a-z, comma, colon, and space, so we
+      # can parse without risking escaping our bash jail
+
+      if [[ ! -z "${buildBotCommand// }" ]]; then
+        _build_bot_update_commands $buildBotCommand
+      fi
+    done
   done
 
+  if [[ ${#prCommands[@]} -gt 0 ]]; then
```

Comment:
I moved the PR parse here, a few lines of code duplicated with minor diffs (ll.106-120); it's not really worth a refactor to remove the duplication IMO.

This refactor here was done to enable detailed logging of source of each build bot command